en/docs/samples/device-based-adaptive-auth.md

```diff
@@ -0,0 +1,111 @@
+# Configure Device-Based Adaptive Authentication
+
+This page guides you through configuring device-based adaptive authentication for a sample web application using a sample hardware key authenticator. 
+
+----
+
+If you have your own application, click the button below.
+
+<a class="samplebtn_a" href="../../guides/configure-adaptive-auth" target="_blank" rel="nofollow noopener">I have my own application</a>
+
+----
+
+## Scenario
+
+The instructions below guide you through specifying authentication steps based on the device that the user uses to log in to the application. In this example, if the user uses a new device or a new browser to log in to the application, the user is prompted for a second step of authentication. The new device or browser is identified by a cookie, therefore, once the cookie expires (this expiry time is specified in the authentication script), the same browser or device is considered as a new device.
+
+----
+
+{!fragments/adaptive-auth-samples.md!}
+
+----
+
+## Configure device-based authentication
+
+1.  Click **Service Providers>List**.
+
+2.  Click **Edit** on the `saml2-web-app-pickup-dispatch.com` service provider.
+
+3.  Expand the **Local and Outbound Configuration** section and click **Advanced Authentication**.
+
+4.  Expand **Script Based Conditional Authentication**.
+
+5.  Click **Templates** on the right side of the **Script Based Conditional Authentication** field and then click **New-Device-Based**. 
+
+    ![device-based-template](../assets/img/samples/device-based-template.png)
+
+6.  Click **Ok**. The authentication script and authentication steps
+    are configured. 
+    
+    The authentication script prompts the second step of authentication if the user logs in using a new device or browser that they have not used before. The new device or browser is identified by a cookie.
+    
+7. Set the `deviceRememberPeriod` to specify how long the device should be considered as a trusted device. 
+
+    Once this time period has elapsed, the device is considered as a new device. The time period is set to two years by default.
+    
+    ``` java
```

Comment:
Remove tab spacing before var deviceRememberPeriod = 60 * 60 * 24 * 365 * 2; . Please fix this in all code blocks.